swap/src/bitcoin/wallet.rs

```diff
@@ -293,6 +294,9 @@ where
         Ok(Amount::from_sat(fees))
     }
 
+    /// Builds a partially signed transaction from the given recipient address
+    /// and amount. Ensures that the address script is at output index `0`
+    /// for the created transaction.
```

Comment:
Nit: The usual convention for rustdoc is to limit the basic explanation to one sentence and follow up with a more elaborate paragraph further down (basically the same as in a commit message).

Source: https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components